lib/agent0/examples/interactive_econ.py

```diff
@@ -0,0 +1,424 @@
+    randseed: int = 0
+
+    def __post_init__(self):
+        """Calculate parameters for the experiment."""
+        self.rate_required_for_same_price: float = min(
+            1, 0.035 * 365 / self.term_days
+        )  # get same price for shorter term
+        self.starting_fixed_rate: FixedPoint = FixedPoint(self.rate_required_for_same_price)  # equivalent of 3.5%
+        self.starting_variable_rate: FixedPoint = FixedPoint(self.rate_required_for_same_price)  # equivalent of 3.5%
+        self.term_seconds: int = 60 * 60 * 24 * self.term_days
+
+
+exp = ExperimentConfig()
+for key in os.environ:
+    if key in fields(exp):
+        setattr(exp, key, os.environ[key])
+rng = np.random.default_rng(seed=exp.randseed)
+
+# %%
+# set up chain
+chain = LocalChain(LocalChain.Config())
+
+# %%
+# Parameters for pool initialization. If empty, defaults to default values, allows for custom values if needed
+config = InteractiveHyperdrive.Config(
+    position_duration=exp.term_seconds,
+    checkpoint_duration=60 * 60 * 24,  # 1 day
+    initial_liquidity=FixedPoint(20),
+    initial_fixed_rate=exp.starting_fixed_rate,
+    initial_variable_rate=exp.starting_variable_rate,
+    curve_fee=exp.curve_fee,
+    flat_fee=exp.flat_fee,
+    governance_fee=exp.governance_fee,
+)
+MINIMUM_TRANSACTION_AMOUNT = config.minimum_transaction_amount
+for k, v in config.__dict__.items():
+    print(f"{k:26} : {v}")
+print(f"{'term length':27}: {exp.term_days}")
+interactive_hyperdrive = InteractiveHyperdrive(chain, config)
+print(f"spot price = {interactive_hyperdrive.hyperdrive_interface.calc_spot_price()}")
+
+# %%
+# set up agents
+larry = interactive_hyperdrive.init_agent(base=FixedPoint(exp.amount_of_liquidity), name="larry")
+larry.add_liquidity(base=FixedPoint(exp.amount_of_liquidity))  # 10 million
+rob = interactive_hyperdrive.init_agent(base=FixedPoint(exp.amount_of_liquidity), name="rob")
+# this verifies that spot price does not change after adding liquidity
+print(f"spot price after adding liquidity = {interactive_hyperdrive.hyperdrive_interface.calc_spot_price()}")
+
+# %%
+# do some trades
+Max = NamedTuple("Max", [("base", FixedPoint), ("bonds", FixedPoint)])
+GetMax = NamedTuple("GetMax", [("long", Max), ("short", Max)])
+
+start_time = time.time()
+
+
+def get_max(
+    _interactive_hyperdrive: InteractiveHyperdrive, _share_price: FixedPoint, _current_base: FixedPoint
+) -> GetMax:
+    """Get max trade sizes.
+
+    Returns
+    -------
+    GetMax
+        A NamedTuple containing the max long in base, max long in bonds, max short in bonds, and max short in base.
+    """
+    max_long_base = _interactive_hyperdrive.hyperdrive_interface.calc_max_long(budget=_current_base)
+    max_long_shares = _interactive_hyperdrive.hyperdrive_interface.calc_shares_out_given_bonds_in_down(max_long_base)
+    max_long_bonds = max_long_shares * _share_price
+    max_short_bonds = FixedPoint(0)
+    try:  # sourcery skip: do-not-use-bare-except
```

Comment:
I'm concerned why you needed this try/except in the first place. Was this failing often? I think if we assume the testing is sufficient then this should alway succeed, right? What sorts of errors were you getting?

Reply:
how's this? I haven't seen any errors arising from this lately.

```
    try:  # sourcery skip: do-not-use-bare-except
        max_short_bonds = _interactive_hyperdrive.hyperdrive_interface.calc_max_short(budget=_current_base)
    except Exception as exc:  # pylint: disable=broad-exception-caught
        print("Error calculating max short bonds: %s. ", exc)
```